openwisp_controller/config/tests/test_admin.py

```diff
@@ -701,10 +705,17 @@ def test_preview_device_jsonerror(self):
 
     def test_preview_device_showerror(self):
         t1 = Template.objects.get(name='dhcp')
-        t2 = Template(name='t', config=t1.config, backend='netjsonconfig.OpenWrt')
-        t2.full_clean()
-        t2.save()
-        templates = [t1, t2]
+        t2 = Template(name='t2', config=t1.config, backend=t1.backend)
+        t3 = Template(name='t3', config=t1.config, backend=t1.backend)
+        t4 = Template(
+            name='t4',
+            config={"interfaces": [{"name": "eth0", "type": "bridge", "stp": "WRONG"}]},
+            backend='netjsonconfig.OpenWrt',
+        )
+        # skip validating config to raise error later
+        t4.save()
+        # adding multiple templates to ensure the order is retained correctly
+        templates = [t1, t2, t3, t4]
```

Comment:
The ValidationError is now raised because of the error in the config and not because of duplicated templates. 

This comment should be updated 

https://github.com/openwisp/openwisp-controller/blob/dd225136bc71ef629c41ed1492078ee0d75395be/openwisp_controller/config/tests/test_admin.py#L728-L730

Reply:
ok, thanks for pointing that out, done: https://github.com/openwisp/openwisp-controller/commit/8ac642b31e9c23ffecd18aacc3c0d9ea134690eb.